python/foglamp/services/south/server.py

```diff
@@ -185,7 +189,9 @@ def __init__(self):
                                                                   timestamp=data['timestamp'],
                                                                   key=data['key'],
                                                                   readings=data['readings']))
-                await asyncio.sleep(sleep_seconds)
+                delta = self._event_loop.time() - t1
+                # If delta somehow becomes > sleep_seconds, then do not sleep
+                await asyncio.sleep(sleep_seconds - delta)
```

Comment:
so what if -ve now? how `asyncio.sleep` behaves then. I see `0` handling is there in source. but how about -ve

Reply:
sleep seconds is always expected as non zero positive integer. Any value <= 0 will simply be ignored but no error is thrown.

Reply:
Try this:
```
import asyncio
def a(tt):
    loop = asyncio.get_event_loop()
    t1 = loop.time()
    loop.run_until_complete(asyncio.sleep(tt))
    t2 = loop.time()
    print(t1, t2, t2-t1)

a(1)
52584.854357898 52585.855137736 1.0007798379956512
a(2)
52589.162718537 52591.165217214 2.0024986770004034
a(-1)
52594.435949676 52594.436174597 0.00022492100106319413
a(-2)
52596.887585229 52596.88778002 0.00019479099864838645
a(-24242323423)
52600.889890059 52600.890121309 0.0002312500000698492
a(0)
52671.16052946 52671.160783865 0.0002544050003052689

```